File: BootCampOOP/Basic/Complex/Complex.cpp
#include "Complex.h"

Complex::Complex(double realIn, double imaginaryIn){
    this->real = realIn;
    this->imaginary = imaginaryIn;
}

Complex::~Complex() {}

double Complex::getReal() const {
    return real;
}

double Complex::getImaginary() const {
    return imaginary;
}

bool Complex::operator==(const Complex& other) const {
    return real == other.getReal() && imaginary == other.getImaginary();
}

bool Complex::operator!=(const Complex& other) const {
    return !operator==(other);
}

void Complex::operator+=(const Complex& other){
    real += other.getReal();
    imaginary += other.getImaginary();
}

void Complex::operator-=(const Complex& other){
    real -= other.getReal();
    imaginary -= other.getImaginary();
}

Complex Complex::operator+(const Complex& other) const {
    Complex result;
    result.real = this->real + other.getReal();
    result.imaginary = this->imaginary + other.getImaginary();
    
    return result;
}

Complex Complex::operator-(const Complex& other) const {
    Complex result;
    result.real = this->real - other.getReal();
    result.imaginary = this->imaginary - other.getImaginary();
    
    return result;
}

Complex Complex::operator*(const Complex& other) const {
    Complex result;
    result.real = this->real * other.getReal();
    result.imaginary = this->imaginary * other.getImaginary();
    
    return result;
}

std::ostream& operator<<(std::ostream& out, const Complex& complex) {
    out << complex.getReal() << (complex.getImaginary() > 0 ? "+" : "") << 
    complex.getImaginary() << "i";
    return out;
};

File: BootCamp Java/Basic/Complex/Complex.java

public class Complex {
    private double real;
    private double imaginary;

    public Complex(double real, double imaginary) {
        this.real = real;
        this.imaginary = imaginary;
    }

    public Complex() {
        this(7, 5);
    }

    public double getReal() {
        return real;
    }
    
    public double getImaginary() {
        return imaginary;
    }

    public boolean isEqual(Complex other) {
        return this.real == other.real && this.imaginary == other.imaginary;
    }
    
    public boolean isNotEqual(Complex other) {
        return !isEqual(other);
    }
    
    public void increment(Complex other) {
        this.real += other.real;
        this.imaginary += other.imaginary;
    }
    
    public void decrement(Complex other) {
        this.real -= other.real;
        this.imaginary -= other.imaginary;
    }
    
    public Complex sum(Complex other) {
        Complex sum = new Complex();
    
    sum.real = this.real + other.real;
    sum.imaginary = this.imaginary + other.imaginary;
    return sum;
    }
    
    public Complex diff(Complex other) {
        Complex diff = new Complex();

        diff.real = this.real - other.real;
        diff.imaginary = this.imaginary - other.imaginary;
        return diff;
    }
    
    public Complex product(Complex other) {
        Complex product = new Complex();

        product.real = this.real * other.real;
        product.imaginary = this.imaginary * other.imaginary;
        return product;
    }

    public String toString() {
        return "(" + real + "+" + imaginary + "im )";
    }
}